tap_google_search_console/client.py

```diff
@@ -1,312 +1,107 @@
+        self.__access_token, self.__expires, self.base_url = None, None, None
         self.__session = requests.Session()
-        self.base_url = None
 
-        # if the 'timeout_from_config' value is 0, "0", "" or not passed then set default value of 300 seconds.
-        if timeout_from_config and float(timeout_from_config):
-            # update the request timeout for the requests
-            self.request_timeout = float(timeout_from_config)
-        else:
-            # set the default timeout of 300 seconds
+        try:
+            self.request_timeout = float(timeout) if (timeout not in (None, 0, "0", "0.0")) else REQUEST_TIMEOUT
+        except ValueError:
             self.request_timeout = REQUEST_TIMEOUT
 
-    def check_sites_access(self):
-        site_list = self.__site_urls.replace(" ", "").split(",")
-
-        for site in site_list:
-            site_encoded = quote(site, safe='')
-
-            #Sample query API for check site access
-            path = 'sites/{}/searchAnalytics/query'.format(site_encoded)
-            body = {'startDate': '2021-04-01', 'endDate': '2021-05-01'}
-            self.post(path=path, data=json.dumps(body))
+    def check_sites_access(self) -> None:
+        """Perform access check for each site url provided"""
+        body = json.dumps({"startDate": "2021-04-01", "endDate": "2021-05-01"})
+        for _ in self.__site_urls.replace(" ", "").split(","):
+            self.post(f"sites/{quote(_, safe='')}/searchAnalytics/query", data=body)
 
-    # during 'Timeout' error there is also possibility of 'ConnectionError',
-    # hence added backoff for 'ConnectionError' too.
-    @backoff.on_exception(backoff.expo,
-                          (Timeout, ConnectionError),
-                          max_tries=5,
-                          factor=2)
+    @backoff.on_exception(backoff.expo, Server5xxError, max_tries=5, factor=2)
     def __enter__(self):
         self.get_access_token()
         return self
 
     def __exit__(self, exception_type, exception_value, traceback):
         self.__session.close()
 
-    @backoff.on_exception(backoff.expo,
-                          Server5xxError,
-                          max_tries=5,
-                          factor=2)
-    def get_access_token(self):
-        # The refresh_token never expires and may be used many times to generate each access_token
-        # Since the refresh_token does not expire, it is not included in get access_token response
-        if self.__access_token is not None and self.__expires > datetime.utcnow():
-            return
-
-        headers = {}
-        if self.__user_agent:
-            headers['User-Agent'] = self.__user_agent
+    def get_access_token(self) -> None:
+        """Performs authentication and the access token if expired"""
 
+        if self.__access_token and self.__expires > datetime.now(timezone.utc):
+            return
+        headers = {"User-Agent": self.__user_agent or ""}
```

Comment:
```suggestion
        headers = {"User-Agent": self.__user_agent} if self.__user_agent or {}
```
if the user_agent is not required

Reply:
Tested without `user-agent` on local, worked fine but not sure if it will have any effect on K8s. So not changing this